pkg/service/service.go

```diff
@@ -1922,7 +1922,7 @@ func (s *Service) restoreServicesLocked(svcBackendsById map[lb.BackendID]struct{
 		// the changed M param.
 		ipv6 := newSVC.frontend.IsIPv6() || (svc.NatPolicy == lb.SVCNatPolicyNat46)
 		recreated := s.lbmap.IsMaglevLookupTableRecreated(ipv6)
-		if option.Config.LoadBalancerOnly && newSVC.useMaglev() && recreated {
+		if newSVC.useMaglev() && recreated {
```

Comment:
it seems like this change is breaking the tests in  `github.com/cilium/cilium/pkg/service`

```
    service_test.go:1513: 
        	Error Trace:	/home/runner/work/cilium/cilium/pkg/service/service_test.go:1513
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestRestoreServiceWithTerminatingBackends
```